ui/lib/utils/tableColumnFactory.tsx

```diff
@@ -0,0 +1,256 @@
+import { Tooltip } from 'antd'
+import { max as _max } from 'lodash'
+import {
+  IColumn,
+  ColumnActionsMode,
+} from 'office-ui-fabric-react/lib/DetailsList'
+import React from 'react'
+import { getValueFormat } from '@baurine/grafana-value-formats'
+
+import {
+  Bar,
+  Pre,
+  TextWithInfo,
+  TextWrap,
+  DateTime,
+  HighlightSQL,
+  IColumnKeys,
+} from '@lib/components'
+
+type Bar = { [key: string]: string }
+type BarsConfig = {
+  displayTransKey?: string // it is same as avg field name default
+  bars: [Bar, Bar, Bar?] // [avg, max, min?]
```

Comment:
Simply `bars: Record<string, string>[]`?

Reply:
Because I want to the bars only have 2 or 3 items, so I didn't use array.